packages/patternfly-4/react-core/src/components/Form/FormGroup.js

```diff
@@ -0,0 +1,134 @@
+          'aria-label',
+          `When label is not a string, at least one of Form Group children should have an aria-label attribute.`
+        );
+      } else if (React.isValidElement(props.label)) {
+        if (!props.children.props['aria-label']) {
+          return new Error(`When label is not a string, a Form Group child should have an aria-label attribute.`);
+        }
+      }
+    } else if (Array.isArray(props.children)) {
+      return childrenArrayContains(props.children, 'id', `At least one of Form Group children should have an id.`);
+    } else if (!props.children.props.id) {
+      return new Error(`A child element of Form Group has to have an id.`);
+    }
+    return null;
+  },
+  /** Additional classes added to the FormGroup. */
+  className: PropTypes.string,
+  /** Label text before the field. */
+  label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+  /** Sets the FormGroup required. */
+  isRequired: PropTypes.bool,
+  /** Sets the FormGroup isValid. */
+  isValid: PropTypes.bool,
+  /** Sets the FormGroup isInline. */
+  isInline: PropTypes.bool,
+  /** Helper text after the field. It can be a simple text, a function or an object. */
+  helperText: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+  /** Helper text after the field when the field is isValid. It can be a simple text, a function or an object. */
+  helperTextInvalid: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+  /** ID of the included field. It has to be the same for proper working. */
+  fieldId: PropTypes.string.isRequired
+};
+
+const defaultProps = {
+  children: null,
+  className: '',
+  label: undefined,
+  isRequired: false,
+  isValid: true,
+  isInline: false,
+  helperText: undefined,
+  helperTextInvalid: undefined,
+  fieldId: undefined
+};
+
+const renderHelperText = ({ helperText, classes, fieldId }) => {
+  if (typeof helperText === 'object') {
+    return helperText;
+  }
+  return (
+    <p className={classes} id={`${fieldId}-helper`} aria-live="polite">
+      {helperText}
+    </p>
+  );
+};
+
+const FormGroup = ({
+  className,
+  children,
+  label,
+  isRequired,
+  isValid,
+  isInline,
+  helperText,
+  helperTextInvalid,
+  fieldId,
+  ...props
+}) => (
+  <FormContext.Consumer>
+    {({ isHorizontal }) => (
+      <div {...props} className={css(styles.formGroup, getModifier(styles, isInline && 'inline'), className)}>
+        {React.isValidElement(label)
+          ? label
+          : typeof label === 'string' && (
+              <label className={css(styles.formLabel)} htmlFor={fieldId}>
+                {label}
+                {isRequired && (
+                  <span className={css(styles.formLabelRequired)} aria-hidden="true">
+                    {ASTERISK}
+                  </span>
+                )}
+              </label>
+            )}
+        {isHorizontal ? <div className={css(styles.formHorizontalGroup)}>{children}</div> : children}
+        {helperText &&
```

Comment:
once the helper text is simplified this can be written as
```jsx
<p 
  className={css(styles.formHelperText, isValid ? styles.modifiers.error : styles.modifiers.info)} 
  id={`${fieldId}-helper`} 
  aria-live="polite"
>
   {isInvalid ? helperTextInvalid ? helperText}
</p>